Homework/ElectricalAppliances/Services/HouseService.cs

```diff
@@ -0,0 +1,137 @@
+                    new ElectricalOutlet(),
+                    new ElectricalOutlet(),
+                    new ElectricalOutlet(),
+                    new ElectricalOutlet()
+                }
+            };
+        }
+
+        public void ProcessHouse()
+        {
+            AddAvailableDevicesToHouse();
+            PlugInAppliances();
+            Console.WriteLine($"\nTotal power consumption: {CalculatePowerConsumption()} watts/h");
+
+            Console.WriteLine("\nSelect what you want to find: Find devices by brand - x; Energy saving devices - y; Portable devices - u;");
+            var criteria = Console.ReadLine()?.Trim();
+
+            if (criteria == "x")
+            {
+                Console.Write("\nEnter the brand you are looking for: ");
+                var brand = Console.ReadLine();
+                var result = myHouse.HouseDevices.FindByBrand(brand);
+                Console.WriteLine($"There are {brand} brand appliances in the house: {string.Join(", ", result.Select(x => x.DeviceName))}");
+            }
+            else if (criteria == "y")
+            {
+                Console.Write("\nEnter the desired energy consumption threshold: ");
+                var energyConsumptionInput = Console.ReadLine();
+
+                if (int.TryParse(energyConsumptionInput, out int energyConsumption))
+                {
+                    var result = myHouse.HouseDevices.FindByEnergyConsumption(energyConsumption);
+                    Console.WriteLine($"\nDevices have been found that satisfy the condition of energy consumption < {energyConsumption}: " +
+                        $"{string.Join(", ", result.Select(x => x.DeviceName))}");
+                }
+            }
+            else if (criteria == "u")
+            {
+                var result = myHouse.HouseDevices.FindByIsPortableProperty();
+                Console.WriteLine($"\nDevices have been found that satisfy the condition of been portable: ");
+            }
+            else
+            {
+                Console.Write("You did not select any types...");
+                return;
+            }
+        }
+
+        private void AddAvailableDevicesToHouse()
+		{
+            foreach (var device in availableDevices)
+            {
+                if (device is Fridge)
+                {
+                    myHouse.HouseDevices.Add((Fridge)device);
+                }
+                else if (device is Dishwasher)
+                {
+                    myHouse.HouseDevices.Add((Dishwasher)device);
+                }
+                else if (device is Vacuum)
+                {
+                    myHouse.HouseDevices.Add((Vacuum)device);
+                }
+                else if (device is AirConditioner)
+                {
+                    myHouse.HouseDevices.Add((AirConditioner)device);
+                }
+                _logger.Log(LogType.Info, "Device added to house model");
+            }
+        }
```

Comment:
Just in case, we can refactor this method with  Generic)

Reply:
Can this be counted as homework on generics? :)